slidechain/export.go

```diff
@@ -90,15 +142,36 @@ func (c *Custodian) pegOutFromExports(ctx context.Context) {
 			}
 
 			log.Printf("pegging out export %x: %d of %s to %s", txid, amounts[i], asset.String(), exporters[i])
-			// TODO(vniu): flag txs that fail with unretriable errors in the db
-			err = c.pegOut(ctx, exporter, asset, int64(amounts[i]), tempID, xdr.SequenceNumber(seqnums[i]))
-			if err != nil {
-				log.Fatalf("pegging out tx: %s", err)
+			var peggedOut PegOutState
+			for i := 0; i < 5; i++ {
+				err = c.pegOut(ctx, exporter, asset, int64(amounts[i]), tempID, xdr.SequenceNumber(seqnums[i]))
+				if err == nil { // successful peg-out
+					peggedOut = PegOutOK
```

Comment:
Nit: handle errors first, and avoid indenting non-error logic. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.